Source/Core/VideoCommon/BPStructs.cpp

```diff
@@ -239,22 +239,24 @@ static void BPWritten(const BPCmd& bp)
     // known for configuring these out-of-range copies.
     u32 copy_width = srcRect.GetWidth();
     u32 copy_height = srcRect.GetHeight();
-    if (srcRect.right > EFB_WIDTH || srcRect.bottom > EFB_HEIGHT)
+    if (u32(srcRect.right) > EFB_WIDTH || u32(srcRect.bottom) > EFB_HEIGHT)
     {
       WARN_LOG_FMT(VIDEO, "Oversized EFB copy: {}x{} (offset {},{} stride {})", copy_width,
                    copy_height, srcRect.left, srcRect.top, destStride);
 
       // Adjust the copy size to fit within the EFB. So that we don't end up with a stretched image,
       // instead of clamping the source rectangle, we reduce it by the over-sized amount.
-      if (copy_width > EFB_WIDTH)
+      if (u32(srcRect.right) > EFB_WIDTH)
       {
-        srcRect.right -= copy_width - EFB_WIDTH;
-        copy_width = EFB_WIDTH;
+        u32 adjustment = srcRect.right - EFB_WIDTH;
+        srcRect.right -= adjustment;
+        copy_width -= adjustment;
```

Comment:
I doubt this happens often (if at all?) but if we had a left position of say EFB_WIDTH - 5.  And a right position of EFB_WIDTH + 10 then this value will be negative which seems wrong.

Instead maybe we should recalculate:  `copy_width = srcRect.GetWidth()`

(same for height)

Reply:
I'm not sure you have done your math correctly. Plugging in the numbers you suggest, I've got:

    EFB_WIDTH = 640
    left = 635
    right = 650
    copy_width = 15
   
    adjustment = 650 - 640 = 10
    right = 650 - 10 = 640
    copy_width = 15 - 10 = 5

And width stays as a positive number.

Reply:
Yeah, my apologies, I must have looked at this too quickly.  You are essentially just clamping the bounds and touching up the dimension.  Only time you'd get negative is if the whole rectangle was out of bounds.

Assuming I'm not having another crazy moment, I do think this would read a bit better:

```
srcRect.right = EFB_WIDTH;
copy_width = srcRect.GetWidth();
```

Reply:
You make a good point, I'm going to rewrite this PR to be easier to read.

Reply:
New version should be much simpler to understand.